dlt/dlt_broker.py

```diff
@@ -166,12 +188,23 @@ def add_context(self, context_queue, filters=None):
                 )
 
                 if not self._recv_filter_set_ack(context_filter_ack_queue, True):
+                    failure_reason = ""
+                    if isinstance(self.msg_handler, DLTMessageHandler):
+                        failure_reason = (
+                            "It's possible that DLTClient client does not start."
+                            " If it's a test case, it might be an error"
+                        )
+                    elif isinstance(self.msg_handler, DLTFileSpinner):
+                        failure_reason = (
+                            f"It's possible that dlt file {self._filename} is empty now. No big issue, "
+                            f"filters would be added once after new message is available in dlt file"
+                        )
```

Comment:
same here. add a `fail_ack_message` in DLTMessageHandler and DLTFileSpinner and avoid using if-else here.

Reply:
i think you miss the responsibility of dlt broker and dlt message handler. Sure, adding DispatcherBase:fail_ack_message() could remove ths if-else in dlt broker, but why?

This failure only means DLTBroker does not receive ack at the point it checks, but it does not mean DispatcherBase fails at all, because ack will be done when next message arrives.

In a word, this is only the assumption of DLTBroker, which does not tell the truth of DispatcherBase and has nothing to do with DispatcherBase